ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/MaximumLedgerTimeService.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.index.v2
+
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.value.Value.ContractId
+import com.daml.logging.LoggingContext
+
+import scala.concurrent.Future
+
+trait MaximumLedgerTimeService {
+
+  /** This method serves two purposes:
+    *   1 - Verify that none of the specified contracts are archived (archival of divulged contracts also count)
+    *   2 - Calculate the maximum ledger time of all the specified contracts (divulged contracts do not contribute)
+    * Important note: existence of the contracts is not checked, only the fact of archival.
+    *
+    * @return NotAvailable, if none of the specified contracts are archived, and all of them are divulged contracts (no ledger-time available)
+    *         NotAvailable, if the specified set is empty.
+    *         Max, if none of the specified contracts are archived, and the maximum ledger-time of all specified non-divulged contracts is known
+    *         Archived, if there was at least one contract specified which is archived (this list is not necessarily exhaustive)
+    */
+  def lookupMaximumLedgerTimeAfterInterpretation(ids: Set[ContractId])(implicit
```

Comment:
For me the `AfterInterpretation` postfix adds more confusion than clarification.  The scaladoc is good but does not mention 'interpretation'.  Maybe just call this `lookupMaximumLedgerTime`

Reply:
I would prefer to keep the AfterInterpretation, because the functionality here relying on the fact, that all the contractIds were able to looked up Active once. See the line starting with Important note. (also this naming was source of some debate last time a well :) ).
Actually I agree that it is not entirely clear what good it is that this is used after interpretation, so I extended the doc there.